packages/data-stores/src/products-list/selectors.ts

```diff
@@ -0,0 +1,7 @@
+import type { State } from './reducer';
+
+export const getState = ( state: State ) => state;
+
+export const getProductsList = ( state: State ) => {
+	return state.productsList;
+};
```

Comment:
Glad this happened, great example of the issue of having a param in the resolver (`type`) but not in the selector. This store will have the bug:

```js
getProductsList( 'jetpack' ) // => jetpack products
getProductsList( 'wpcom' ) // => wpcom products
getProductsList( 'jetpack' ) // => wpcom products (oops)
```
Because the store thinks the `jetpack` call is cached, it won't trigger the resolver, meaning you'll get whatever state that's already there. To fix this, I'd make the state keyed by type, like such:
```js
{
    jetpack: ...
    wpcom: ...
}
```
Then in the selector use the `type` prop to get the desired part of the state:

```suggestion
export const getProductsList = ( state: State, type: string ) => {
	return state.productsList[type];
};
```

Reply:
Oh, right! You specifically said that in our call and I didn't pick up on how it applied here. Thank you 🤦‍♂️!

Reply:
That's not really how it's setup. 

The `type` variable is used to filter the product list in the resolver. Then the filtered product list is added to the state. If no type is specified, then the full product list is added to the state.

So the state isn't grouped by type.

Reply:
Oh I see. I see now, it seems to me the resolver isn't using the `type` variable itself, it's only forwarding it to the `receiveProductsList` action creator, and then it isn't used in the reducer. I think we can delete the `type` param from the resolver.

Reply:
Ah, yes! The type variable definitely isn't needed. I'll get that removed and then I think we're good.